scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontendState.java

```diff
@@ -771,10 +803,54 @@ private void dispatchJobStateUpdated(String owner, NotificationData<JobInfo> not
     }
 
     /**
+     * Dispatch the job state updated event
+     * 
+     * @param owner
+     *            the owner of this job
+     * @param notification
+     *            the data to send to every client
+     */
+    private void dispatchJobUpdatedFullData(JobState job) {
+        try {
+            if (logger.isDebugEnabled()) {
+                jlogger.debug(job.getJobInfo().getJobId(), " event [" + SchedulerEvent.JOB_UPDATED + "]");
+            }
+            for (ListeningUser listeningUserId : identifications.values()) {
+                // if this user has a listener
+                if (listeningUserId.isListening()) {
+                    try {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract this nested try block into a separate method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1141)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS134)